src/Microsoft.VisualStudio.Threading/TplExtensions.cs

```diff
@@ -108,6 +108,26 @@ public static async Task<T> WithTimeout<T>(this Task<T> task, TimeSpan timeout)
             return task.GetAwaiter().GetResult();
         }
 
+        /// <summary>
+        /// Gets the <see cref="Task{TResult}.Result"/> of a completed <see cref="Task{TResult}"/>.
+        /// </summary>
+        /// <typeparam name="TResult">The type of the result produced by the <see cref="Task{TResult}"/>.</typeparam>
+        /// <param name="task">The completed task.</param>
+        /// <returns>The result value of the <see cref="Task{TResult}"/>.</returns>
+        /// <exception cref="ArgumentNullException">If <paramref name="task"/> is <see langword="null"/>.</exception>
+        /// <exception cref="InvalidOperationException">If <paramref name="task"/> is not completed.</exception>
```

Comment:
Also `AggregateException` if the Task has faulted, or `OperationCanceledException`(?) if the task was canceled.

Unless you change the exception type thrown by switching to `GetAwaiter().GetResult()`